### PR TITLE
Added error messages to GA error events

### LINKED ISSUE #148: Add message error to GA error events

### PR DESCRIPTION
**Pull request by BrunoRosendo**
closes #148

### CODE REVIEW

File: src/services/companyOffersService.js [hunk 1]
```diff
@@ -35,12 +36,15 @@ export const fetchCompanyOffers = measureTime(TIMED_ACTIONS.COMPANY_OFFERS_FETCH
             return json;
         } catch (error) {
 
-            if (!isErrorRegistered) createEvent(EVENT_TYPES.ERROR(
+            const errorArray = Array.isArray(error) ? error :
+                [{ msg: "Unexpected Error. Please try again later." }];
+
+            if (!isErrorRegistered) createErrorEvent(
```
Comment (imnotteixeira) [inline src/services/companyOffersService.js:42]: when the function call spans over multiple lines, it's better to have the `{}` around the if body

File: src/services/offerService.js [hunk 2]
```diff
@@ -89,13 +94,16 @@ export const hideOffer = measureTime(TIMED_ACTIONS.OFFER_HIDE, async (offerId) =
 
     } catch (error) {
 
-        if (!isErrorRegistered) createEvent(EVENT_TYPES.ERROR(
+        const errorArray = Array.isArray(error) ? error :
+            [{ msg: "Unexpected Error. Please try again later." }];
+
+        if (!isErrorRegistered) createErrorEvent(
```
Comment (imnotteixeira) [inline src/services/offerService.js:100]: when the function call spans over multiple lines, it's better to have the `{}` around the if body

File: src/services/offerService.js [hunk 3]
```diff
@@ -130,13 +139,16 @@ export const disableOffer = measureTime(TIMED_ACTIONS.OFFER_DISABLE, async (offe
...
```
Comment (imnotteixeira) [inline src/services/offerService.js:145]: when the function call spans over multiple lines, it's better to have the `{}` around the if body

File: src/services/offerService.js [hunk 4]
```diff
@@ -167,13 +180,16 @@ export const enableOffer = measureTime(TIMED_ACTIONS.OFFER_ENABLE, async (offerI
...
```
Comment (imnotteixeira) [inline src/services/offerService.js:186]: when the function call spans over multiple lines, it's better to have the `{}` around the if body

File: src/services/offerService.js [hunk 5]
```diff
@@ -248,13 +265,17 @@ export const newOffer = measureTime(TIMED_ACTIONS.OFFER_CREATE, async ({
         return json;
 
     } catch (error) {
-        if (!isErrorRegistered) createEvent(EVENT_TYPES.ERROR(
+
+        const errorArray = Array.isArray(error) ? error :
+            [{ msg: "Unexpected Error. Please try again later." }];
+
+        if (!isErrorRegistered) createErrorEvent(
```
Comment (imnotteixeira) [inline src/services/offerService.js:272]: when the function call spans over multiple lines, it's better to have the `{}` around the if body

File: src/services/offerService.js [hunk 6]
```diff
@@ -325,12 +347,16 @@ export const editOffer = measureTime(TIMED_ACTIONS.OFFER_EDIT, async ({
...
```
Comment (imnotteixeira) [inline src/services/offerService.js:354]: when the function call spans over multiple lines, it's better to have the `{}` around the if body

File: src/utils/analytics/index.js [hunk 7]
```diff
@@ -60,14 +61,32 @@ export const recordTime = (action = TIMED_ACTIONS.UNKNOWN, t0, t1, label) => {
...
+ * Constructs an error event and sends it to Google Analytics
+ *
+ * @param {*} metricId Identifier of the metric where the error occurred
+ * @param {*} type Type of error
+ * @param {*} errors Array of error objects with a 'msg' property defined
+ * @param {*} status HTTP status
+ */
+export const createErrorEvent = (metricId, type, errors, status = 500) => {
+    const label = errors.reduce(
+        (acc, err) => err.param ? `${acc}: ${err.param}-${err.msg}` : `${acc}: ${err.msg}`,
```
Comment (imnotteixeira) [inline src/utils/analytics/index.js:83]:
```suggestion
        (acc, err) => err.param ? `${acc}; ${err.param}-${err.msg}` : `${acc}: ${err.msg}`,
```

**Review by imnotteixeira — CHANGES_REQUESTED** [review]

File: src/utils/analytics/index.js (hunk 7 above)
Comment (BrunoRosendo) [reply src/utils/analytics/index.js:83]: I'll try to put it as `UNEXPECTED_ERROR: error1; error2` then

File: src/components/HomePage/SearchResultsArea/Offer/OfferWidget.js [hunk 8]
```diff
@@ -64,7 +65,7 @@ const OfferWidget = ({
                 visibilityError = getHumanError(err[0]);
 
             addSnackbar({
-                message: visibilityError ? visibilityError : "Unexpected Error. Please try again later.",
+                message: visibilityError ? visibilityError : Constants.UNEXPECTED_ERROR_MESSAGE,
```
Comment (imnotteixeira) [inline src/components/HomePage/SearchResultsArea/Offer/OfferWidget.js:68]:
```suggestion
                message: visibilityError || Constants.UNEXPECTED_ERROR_MESSAGE,
```

**Review by imnotteixeira — CHANGES_REQUESTED** [review]

**Review by imnotteixeira — APPROVED** [review]

**Pull request merged**